### PR TITLE
feat(rns): handles confirmations for buying domains

### PR DESCRIPTION
**Pull request by farinalvaro**
Related to #607 

- [x] track buy domain confirmations
- [x] show loader on buy screen
- [x] show loader on your domains screen
- [x] remove done page, move functionality to  checkout page
- [x] reduce complexity of checkout page component

### CODE REVIEW

File: src/components/pages/rns/RnsRoutes.tsx [hunk 1]
```diff
@@ -33,19 +34,20 @@ const TABS: StyledNavTabProps[] = [
...
-const RnsRoutes = () => {
+const RnsRoutes: FC = () => {
   const { pathname } = useLocation()
 
   const { services } = networkConfig
   const rnsEnabled = services && services.rns
   const history = useHistory()
 
   useEffect(() => {
+    // TODO: wrap this code block to be excecuted only on debugging
```
Comment (jurajpiar) [inline src/components/pages/rns/RnsRoutes.tsx:45]: why did you remove this? I see no issue created for it and I'd rather not create an unused listener in production.

**Review by jurajpiar — CHANGES_REQUESTED** [review]
LG except for my comments and queries bellow and that the complexity of the domains checkout component is still above max allowed. 
```
Arrow function has a complexity of 13. Maximum allowed is 11
```

**Comment by jurajpiar** [comment]
Also, created an issue to refactor the confirmations interface as I believe the actions passed can be done neater: https://github.com/rsksmart/rif-marketplace-ui/issues/692
Please comment there, if you've got any.

File: src/components/pages/rns/RnsRoutes.tsx (hunk 1 above)
Comment (farinalvaro) [reply src/components/pages/rns/RnsRoutes.tsx:45]: Sorry, I didn't get your point. 
I only added a comment that this might not be needed in production. It subscribes to routes and do nothing in production environment.

File: src/components/pages/rns/buy/DomainOffersPage.tsx [hunk 2]
```diff
@@ -142,14 +147,39 @@ const DomainOffersPage: FC = () => {
...
+          <ShortenTextTooltip
+            value={domainName || pseudoResolvedName}
+            maxLength={30}
+          />
+        )
         : <AddressItem pretext="Unknown RNS:" value={tokenId} />
 
+      const isProcessingConfs = buyingConfs.some(
+        ({ contractActionData }) => (
+          (contractActionData as BuyDomainContractData).tokenId === tokenId
```
[the remark this answers is not in the document]
Comment (farinalvaro) [reply src/components/pages/rns/buy/DomainOffersPage.tsx:165]: I agree this can be a problem but I think we could add this to the refactor confirmations issue.
The fact is that `contractActionData` can be of different types: https://github.com/rsksmart/rif-marketplace-ui/pull/686/files#diff-f0bbb412af2adbf50c5954f916931157b8cc9fa0a68deccf8c784d43621a1112R39
But when it comes to buying confirmations it should be of type BuyDomainContractData. https://github.com/rsksmart/rif-marketplace-ui/pull/686/files#diff-72810737a6926f4d5279d4545e16ffb3f4a7dc42ecebc7174cb233e835885cceR242
So that should be the issue we may need, to link actions with proper contract data, similar to what we do with the reducers actions

File: src/components/pages/rns/buy/DomainOffersPage.tsx [hunk 3]
```diff
@@ -68,6 +71,8 @@ const DomainOffersPage: FC = () => {
     },
   } = useContext(Web3Store)
 
+  const buyingConfs = useConfirmations(['RNS_BUY'])
```
[the remark this answers is not in the document]
Comment (farinalvaro) [reply src/components/pages/rns/buy/DomainOffersPage.tsx:74]: If you don't mind I would like to apply this renaming on the next PR so it doesn't create a conflict https://github.com/rsksmart/rif-marketplace-ui/pull/691/files#diff-c26b69f9ea94012b3d9d934b0dbc3211e12f66a63c85d0baea2f66f1ca38b36aR74 😄

File: src/components/pages/rns/buy/DomainOffersPage.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (jurajpiar) [reply src/components/pages/rns/buy/DomainOffersPage.tsx:165]: let's handle it in the confirmation issue then

File: src/components/pages/rns/buy/DomainOffersPage.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (jurajpiar) [reply src/components/pages/rns/buy/DomainOffersPage.tsx:74]: as you wish

**Review by jurajpiar — APPROVED** [review]
